src/app/login/login-form/login-form.component.html

```diff
@@ -0,0 +1,63 @@
+<form name="form" [formGroup]="loginForm" (ngSubmit)="login()">
+  <div class="field">
+    <p class="control has-icons-left has-icons-right">
+      <input
+        class="input"
+        type="text"
+        placeholder="Username"
+        name="username"
+        formControlName="username"
+        required
+      />
+      <span style="color:red" *ngIf="f.submitted && !username.valid">
+        <span *ngIf="submitted && f.username.errors"
+          >Username is required.</span
+        >
+      </span>
+      <span class="icon is-small is-left">
+        <i
+          class="fa fa-envelope"
+          style="font-size: 28px; color: rgb(85, 85, 85);"
+        ></i>
+      </span>
+    </p>
+  </div>
+  <div class="field">
+    <p class="control has-icons-left">
+      <input
+        class="input"
+        type="password"
+        placeholder="Password"
+        name="password"
+        formControlName="password"
+        required
+      />
+      <span style="color:red" *ngIf="f.submitted && !password.valid">
+        <span *ngIf="submitted && f.password.errors"
+          >Password is required.</span
+        >
+      </span>
+      <span class="icon is-small is-left">
+        <i
+          class="fa fa-link"
+          style="font-size: 28px; color: rgb(87, 83, 83);"
+        ></i>
+      </span>
+    </p>
+  </div>
+  <div class="field">
+    <span style="color: red;" *ngIf="submitted && error">{{ error }}</span>
+  </div>
+  <div class="field">
+    <p class="control">
+      <button [disabled]="loading" class="button is-success">
+        Login
+      </button>
+      <img
+        *ngIf="loading"
+        src="data:image/gif;base64,R0lGODlhEAAQAPIAAP///wAAAMLCwkJCQgAAAGJiYoKCgpKSkiH/C05FVFNDQVBFMi4wAwEAAAAh/hpDcmVhdGVkIHdpdGggYWpheGxvYWQuaW5mbwAh+QQJCgAAACwAAAAAEAAQAAADMwi63P4wyklrE2MIOggZnAdOmGYJRbExwroUmcG2LmDEwnHQLVsYOd2mBzkYDAdKa+dIAAAh+QQJCgAAACwAAAAAEAAQAAADNAi63P5OjCEgG4QMu7DmikRxQlFUYDEZIGBMRVsaqHwctXXf7WEYB4Ag1xjihkMZsiUkKhIAIfkECQoAAAAsAAAAABAAEAAAAzYIujIjK8pByJDMlFYvBoVjHA70GU7xSUJhmKtwHPAKzLO9HMaoKwJZ7Rf8AYPDDzKpZBqfvwQAIfkECQoAAAAsAAAAABAAEAAAAzMIumIlK8oyhpHsnFZfhYumCYUhDAQxRIdhHBGqRoKw0R8DYlJd8z0fMDgsGo/IpHI5TAAAIfkECQoAAAAsAAAAABAAEAAAAzIIunInK0rnZBTwGPNMgQwmdsNgXGJUlIWEuR5oWUIpz8pAEAMe6TwfwyYsGo/IpFKSAAAh+QQJCgAAACwAAAAAEAAQAAADMwi6IMKQORfjdOe82p4wGccc4CEuQradylesojEMBgsUc2G7sDX3lQGBMLAJibufbSlKAAAh+QQJCgAAACwAAAAAEAAQAAADMgi63P7wCRHZnFVdmgHu2nFwlWCI3WGc3TSWhUFGxTAUkGCbtgENBMJAEJsxgMLWzpEAACH5BAkKAAAALAAAAAAQABAAAAMyCLrc/jDKSatlQtScKdceCAjDII7HcQ4EMTCpyrCuUBjCYRgHVtqlAiB1YhiCnlsRkAAAOwAAAAAAAAAAAA=="
```

Comment:
What is this? Is there a way to make this more readable for what it is/does?

Reply:
It's a loading circle animation. It can be removed if you desire; won't hurt my feelings. If a user has a slow connection it's a signal that things are still working, however, for our task it isn't even seen because login works so fast.